packages/ai-claude-code/src/node/claude-code-service-impl.ts

```diff
@@ -0,0 +1,436 @@
+                    };
+                    await receivedResult;
+                })(),
+                options: <typeof Options>{
+                    ...request.options,
+                    abortController,
+                    canUseTool: (toolName: string, toolInput: unknown) => this.requestToolApproval(streamId, toolName, toolInput),
+                    env: { ...process.env, ANTHROPIC_API_KEY: apiKey, NODE_OPTIONS: '' },
+                    stderr: (data: unknown) => this.logger.error('Claude Code Std Error:', data),
+                }
+            });
+
+            for await (const message of stream) {
+                this.client.sendToken(streamId, message);
+                if (message.type === 'result' || abortController.signal.aborted) {
+                    done();
+                    break;
+                }
+            }
+            // Signal stream completion by returning undefined
+            abortController.abort('closed after result');
+            this.client.sendToken(streamId, undefined);
+        } catch (e) {
+            this.logger.error('Claude Code error:', e);
+            this.client.sendError(streamId, e instanceof Error ? e : new Error(String(e)));
+        } finally {
+            this.abortControllers.delete(streamId);
+        }
+    }
+
+    /**
+     * Dynamically imports the Claude Code SDK from the local installation.
+     * @param customClaudeCodePath Optional custom path to Claude Code executable (cli.js)
+     * @returns An object containing the SDK's query function, user message type, and options type.
+     */
+    protected async importClaudeCodeSDK(customClaudeCodePath?: string): Promise<{ query: unknown; SDKUserMessage: unknown; Options: unknown }> {
+        let claudeCodePath: string;
+
+        if (customClaudeCodePath) {
+            if (!existsSync(customClaudeCodePath)) {
+                throw new Error(`Specified Claude Code executable not found at: ${customClaudeCodePath}`);
+            }
+            const realPath = realpathSync(customClaudeCodePath);
+            // Use the directory containing the cli.js file
+            claudeCodePath = path.dirname(realPath);
+        } else {
+            claudeCodePath = this.resolveClaudeCodePath();
+        }
+
+        const sdkPath = path.join(claudeCodePath, 'sdk.mjs');
+
+        // Check if file exists before importing
+        if (!existsSync(sdkPath)) {
+            throw new Error('Claude Code installation not found. ' +
+                'Please install with: `npm install -g @anthropic-ai/claude-code` ' +
+                'and/or specify the path to the executable in the settings. ' +
+                `We looked at ${sdkPath}`);
+        }
+
+        // Use eval to hide the dynamic import from Webpack's static analysis
+        const importPath = `file://${sdkPath}`;
+        // eslint-disable-next-line no-eval
+        const dynamicImport = (0, eval)('(path) => import(path)');
+        return dynamicImport(importPath);
```

Comment:
This seems very hacky, surely there is a cleaner workaround or proper webpack config